solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -258,5 +260,13 @@ def process_forecast_observations(forecast_observations, filters, data,
                 preprocessing_results=preproc_results,
                 forecast_values=forecast_values,
                 observation_values=observation_values)
-            processed_fxobs.append(processed)
-    return processed_fxobs
+            processed_fxobs[name] = processed
+    return tuple(processed_fxobs.values())
+
+
+def _name_pfxobs(current_names, forecast_name, i=1):
+    if forecast_name in current_names:
+        new_name = f'{forecast_name}-{i}'
+        return _name_pfxobs(current_names, new_name, i + 1)
```

Comment:
love an opportunity for infinite recursion. probably ok here. an alternative is something like `for _ in range(max_i)`

Reply:
ok, it's limited. Also changed the name format. before it was making names like 'name-1-2'